lumigator/frontend/src/stores/datasets/store.js

```diff
@@ -4,14 +4,45 @@ import datasetsService from "@/services/datasets/datasetsService";
 
 export const useDatasetStore = defineStore('dataset', () => {
   const datasets = ref([]);
+  const selectedDataset = ref(null);
 
 
   async function loadDatasets() {
-    if (datasets.value.length > 0) { return; }
     datasets.value = await datasetsService.fetchDatasets();
   }
+
+  async function loadDatasetInfo(datasetID) {
+    selectedDataset.value = await datasetsService.fetchDatasetInfo(datasetID);
+  }
+
+  async function uploadDataset(datasetFile) {
+    if (!datasetFile) { return }
+    // Create a new FormData object and append the selected file and the required format
+    const formData = new FormData();
+    formData.append('dataset', datasetFile); // Attach the file
+    formData.append('format', 'job'); // Specification @localhost:8000/docs
+    const uploadConfirm = await datasetsService.postDataset(formData)
+    if (uploadConfirm.status) {
+      console.log('⚠️ Error', uploadConfirm.message);
+    }
+    await loadDatasets();
+  }
+
+  async function deleteDataset(id) {
+    if (!id) { return };
+    if (selectedDataset.value && id === selectedDataset.value.id) {
+      selectedDataset.value = null
+    }
+    await datasetsService.deleteDataset(id);
+    await loadDatasets();
+  }
+
```

Comment:
Is this out of date/duplicated from: https://github.com/mozilla-ai/lumigator/blob/39d3d67ab55c5148c395fb473f641c9c0f798433/lumigator/frontend/src/components/organisms/LUpload.vue#L48-L73?

Reply:
This component was deleted, probably came back to life after rebasing 
🧹